fix FreeBSD support

Erk- opened the pull request:
### Checklist

* [x] I have read the [Contributor Guide](../../CONTRIBUTING.md)
* [x] I have read and agree to the [Code of Conduct](../../CODE_OF_CONDUCT.md)
* [x] I have added a description of my changes and why I'd like them included in the section below

### Description of Changes

The libraries needed for FreeBSD support are not in the default
search-path on the system, this patch adds /usr/local/lib to rustc's
search path on FreeBSD.

### Related Issues

List related issues here

Erk- committed:
fix FreeBSD support

The libraries needed for FreeBSD support are not in the default
search-path on the system, this patch adds /usr/local/lib to rustc's
search path on FreeBSD.

Signed-off-by: Valdemar Erk <valdemar@erk.io>

Jake-Shadle commented:
I've never really had to deal with FreeBSD before, is there no environment variable or pkg-config like utility to locate the path for libraries when linking? I mean if it works that's fine, just seems odd to have a hardcoded path.

Erk- commented:
Would it be better to change it to use something like:
```shell
(21:25:20 </usr/home/erk/dev/nfd2>) 0 $ pkg-config --libs-only-L gtk+-3.0 glib-2.0
-L/usr/local/lib
```

Jake-Shadle commented:
Yes please! That's what the linux one does so having something close to that that works for the BSD case would be great.

Erk- committed:
Made the implementation more robust

The build script now uses pkg-config to generate the list of
directonaries needed in the search path.

Signed-off-by: Valdemar Erk <valdemar@erk.io>

Jake-Shadle commented:
(also you can rebase off main to get the CI fix)

Jake-Shadle reviewed (approved):
Great, thanks for the PR!
